cpp/CMakeLists.txt

```
@@ -137,12 +127,56 @@ Provide targets for KvikIO: C++ bindings for cuFile.
 ]=]
 )
 
+set(final_code_string
+    [=[
+get_property(already_set_kvikio DIRECTORY PROPERTY kvikio_already_set_defines SET)
+if(NOT already_set_kvikio)
+  set_property(DIRECTORY PROPERTY kvikio_already_set_defines "ON")
+
+  # Find cuFile and determine which features are supported
+  find_package(cuFile)
+  if(NOT cuFile_FOUND)
+    message(WARNING "KvikIO: Building KvikIO without cuFile")
```

It might be helpful to to consumers to word this warning and other messages below such that it's clear that the warning comes from using kvikio headers when building their own library (as opposed to kvikio being built by CPM, for instance).

---

Can you suggest something specific? I don't understand your goal.

---

The messages here begin with `KvikIO: ` so it should be clear where this comes from.

---

Hmm I was thinking of trying to differentiate building kvikio itself from building some pkg against kvikio, but I am also struggling to come up with good wording. Perhaps something like this?
```suggestion
    message(WARNING "${PROJECT_NAME}: Using KvikIO headers without cuFile")
```

---

This will never be called except when building some package against kvikio. I think the current message is clear enough and don't really want to do variable injection here...

---

The messages are immediately followed by the CPM message about finding kvikio, so it's clear that it's happening during the find_package call.

---

I don't follow why you're concerned about "variable injection" here. There's nothing magical here.

The confusing case will be if CPM is adding kvikio from source instead of being found as an installed package, in which case you'll see any messages that occur as part of kvikio's build as well as these messages that occur when the built kvikio is found.